### PR TITLE
Tranquilizer.flush() to make batch / micro-batch oriented workflows easier

### LINKED ISSUE #85: SimpleTranquilizerAdapter.flush() should force a flush even if lingerMillis has not elapsed
Currently, flush() doesn't actively force Tranquility to send buffered messages to Druid but rather blocks until lingerMillis elapses and the send thread does its thing. This means that the method will block for up to lingerMillis + sendTime when it's called. It would be great if calling flush() would immediately begin flushing the buffer regardless of what lingerMillis is set to.

### PR DESCRIPTION
**Pull request by gianm**
Fixes #85

### CODE REVIEW

**Comment by fjy** [comment]
:+1:

File: core/src/main/scala/com/metamx/tranquility/tranquilizer/Tranquilizer.scala [hunk 1]
```diff
@@ -62,13 +62,16 @@ class Tranquilizer[MessageType] private(
 
   case class MessageHolder(message: MessageType, future: Promise[Unit])
 
-  // lock synchronizes access to buffer, bufferStartMillis, pendingBatches.
+  // lock synchronizes access to buffer, bufferStartMillis, pendingBatches, currentBatch*, flushing.
```
Comment (xvrl) [inline core/src/main/scala/com/metamx/tranquility/tranquilizer/Tranquilizer.scala]: `*` are cool, but what do they mean

Comment (gianm) [inline core/src/main/scala/com/metamx/tranquility/tranquilizer/Tranquilizer.scala]: @xvrl pushed some more comments

**Comment by dclim** [comment]
:+1:

**Pull request merged**